src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.ts

```diff
@@ -40,13 +41,15 @@ export enum CollapsableCodeEditorElement {
 })
 export class CodeEditorContainerComponent implements OnChanges, ComponentCanDeactivate {
     readonly CommitState = CommitState;
+    readonly EditorState = EditorState;
     readonly CollapsableCodeEditorElement = CollapsableCodeEditorElement;
     @ViewChild(CodeEditorGridComponent, { static: false }) grid: CodeEditorGridComponent;
 
     @ViewChild(CodeEditorFileBrowserComponent, { static: false }) fileBrowser: CodeEditorFileBrowserComponent;
     @ViewChild(CodeEditorActionsComponent, { static: false }) actions: CodeEditorActionsComponent;
     @ViewChild(CodeEditorBuildOutputComponent, { static: false }) buildOutput: CodeEditorBuildOutputComponent;
     @ViewChild(CodeEditorAceComponent, { static: false }) aceEditor: CodeEditorAceComponent;
```

Comment:
This is now possibly undefined (if the other editor gets used)
```suggestion
    @ViewChild(CodeEditorAceComponent, { static: false }) aceEditor?: CodeEditorAceComponent;
```

Reply:
Two other components accessed members of this attribute, so I had to change those as well. (These other components will be replaced next and currently use only ace, so those undefined checks are only to appease the typescript compiler 😅 ).